Cargo.toml

```diff
@@ -4,7 +4,7 @@ members = ["crates/*"]
 [workspace.package]
 version = "0.45.0"
 edition = "2021"
-rust-version = "1.64"  # Keep in sync with README.md and rust-toolchain.toml.
+rust-version = "1.73"  # Keep in sync with README.md and rust-toolchain.toml.
```

Comment:
This is a huge MSRV bump, to the very latest release of Rust.  Is it absolutely required?  If so, I think we're going to have to find an alternative to `faer`.

Reply:
I don't know. I didn't choose the version. Some tool told me to use it when compilation failed. I don't recall which. I'm unable to find any mention of an MSRV in the `faer` repo. I'll find out.

Reply:
The author of faer offered to modify the library to require 1.65 if we want to use it.

Reply:
Raise faer version to 0.13.5, lower rustc to 1.67.0 in e61bb4d

Reply:
Presently, the CI for `faer` uses `rustc` 1.67. The Qiskit MSRV is 1.7.0. So this conversation is resolved.